### ISSUE TITLE
Negative emissions farm EmissionPenalty

### ISSUE THREAD

**Issue by willu47** [OSeMOSYS/OSeMOSYS_GNU_MathProg#99]
@Timon-R has identified that use of EmissionPenalty subsidises negative emissions, resulting in energy system models "farming" negative emissions to reduce the total energy system costs.

@FraGard suggests the following solution modifying the E5 constraint to only be active for positive values of `EmissionAcitivityRatio`, e.g. adding `: EmissionActivityRatio > 0`

Change:
```ampl
s.t. E5_DiscountedEmissionsPenaltyByTechnology{r in REGION, t in TECHNOLOGY, y in YEAR}: sum{e in EMISSION, l in TIMESLICE, m in MODE_OF_OPERATION} EmissionActivityRatio[r,t,e,m,y]*RateOfActivity[r,l,t,m,y]*YearSplit[l,y]*EmissionsPenalty[r,e,y]/DiscountFactorMid[r,y] = DiscountedTechnologyEmissionsPenalty[r,t,y];
```
To:
```ampl
s.t. E5_DiscountedEmissionsPenaltyByTechnology{r in REGION, t in TECHNOLOGY, y in YEAR}: sum{e in EMISSION, l in TIMESLICE, m in MODE_OF_OPERATION: EmissionActivityRatio > 0} EmissionActivityRatio[r,t,e,m,y]*RateOfActivity[r,l,t,m,y]*YearSplit[l,y]*EmissionsPenalty[r,e,y]/DiscountFactorMid[r,y] = DiscountedTechnologyEmissionsPenalty[r,t,y];
```

**Comment by HauHe** [comment]
I think the suggested solution might to lead to distorted results in relation to negative emissions, i.e., models will underestimate their role. This problem is for me a reason why I prefer to work with emission limits since their positive and negative emissions balance out, as they would in an emission trading system. Where negative emissions will start playing a role once the price for certificates is high enough for negative emission techs to break even. 
But I see the need to be able to work with an emission penalty since it is required in many projects and model comparisons. However, as said above this formulation might lead to results that underestimate the role of negative-emission-techs.

**Comment by Timon-R** [comment]
I agree with @HauHe and have some additional thoughts on this matter:

1. If emissions penalties are calculated at the technology level, as above, negative emissions activities, such as for CCS technologies, are not rewarded at all, they have the same result (in terms of cost) as if CCS wasn't deployed, i.e. it will never be deployed because non-CCS technologies are cheaper (emissions from fossil fuels are calculated at extraction/import, and CCS technologies then have a negative emissions activity to account for the captured carbon). This seems unrealistic. Therefore, I think it would be better to calculate the emissions penalty more globally, on the total emissions of the model. This doesn't allow for technology-specific costs of emissions, but it would allow negative emissions to count, and account for some of the positive emissions, as in a real carbon credit scheme.
2. Taking the emissions penalty into account globally (including the limit set above that anything less than 0 doesn't count) would also allow the limit to be set below 0. This would have a similar effect to a carbon cap but allows the model to be more flexible. This threshold, below which emissions are penalised/rewarded, could allow emissions from non-electricity sectors to be taken into account. I imagine it could be set as a parameter.

So much about my ideas, curious about what you think about them.
